grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -514,7 +514,23 @@ class StockMovementService {
         List<StockMovementItem> stockMovementItems = []
 
         if (stepNumber == '3') {
-          return getEditPageItems(requisition, max, offset)
+            List editPageItems = getEditPageItems(requisition, max, offset)
+            if (requisition && requisition.sourceType == RequisitionSourceType.ELECTRONIC) {
+                editPageItems = editPageItems.collect { editPageItem ->
+                    // origin = fulfilling, destination = requesting
+                    def quantityOnHandRequesting = productAvailabilityService.getQuantityOnHand(editPageItem.product, requisition.destination)
+                    editPageItem << [quantityOnHandRequesting: quantityOnHandRequesting]
+                    if (requisition.requisitionTemplate) {
+                        def stocklist = Requisition.get(requisition.requisitionTemplate.id)
+                        def quantityOnStocklist = stocklist.requisitionItems?.find { it.product == editPageItem.product && it.orderIndex * 100 == editPageItem.sortOrder }?.quantity?:0
+                        editPageItem << [quantityOnStocklist: quantityOnStocklist]
+                    } else {
+                        def quantityDemand = forecastingService.getDemand(requisition.destination, editPageItem.product)?.totalDemand?:0
+                        editPageItem << [quantityDemand: quantityDemand]
+                    }
+                }
+            }
+            return editPageItems
```

Comment:
Shouldn't we include those fields in the view? Isn't performance too slow with this solution?

Reply:
@DominikaHatala I am not sure if including this in view will improve performance. Because I think that I'd need to join:
- requisition items via requisitionTemplate (from requisition) for stocklist quantity,
- product availability for qoh on requesting (destination) side, 
- product demand details.
I could investigate it (but I am not sure if I will have time for it in the current ticket).
@jmiranda what do you think about it?